Jenkinsfiles/Jenkinsfile.build_app_ami

```diff
@@ -0,0 +1,178 @@
+def private_key = ''
+def vault_pass = 'vault-pass'
+def aws_creds = 'cbj-deploy'
+def release_version = ''
+def slack_channel = '#bluebutton-alerts'
+def BB20_PLATINUM_AMI_ID = ''
+
+pipeline {
+  agent {
+    node {
+      label ''
+      customWorkspace 'blue-button-build-ami'
+    }
+  }
+
+  parameters {
+    string(
+      defaultValue: "",
+      description: 'The branch of the application repo to build. Required.',
+      name: 'BB20_APP_VERSION'
+    )
+    string(
+      defaultValue: "*/master",
+      description: 'The branch of the deployment repo to use for the build.',
+      name: 'BB20_DEPLOY_BRANCH'
+    )
+    string(
+      defaultValue: "",
+      description: 'The AMI ID of the Platinum Gold Image. If unset, the latest will be pulled from AWS and used.',
+      name: 'BB20_PLATINUM_AMI_ID'
+    )
+    string(
+      defaultValue: "",
+      description: 'The subnet ID where the build server will be launched.',
+      name: 'SUBNET_ID'
+    )
+    string(
+      defaultValue: "m3.medium",
+      description: 'The class/size of the ec2 instance to launch.',
+      name: 'INSTANCE_CLASS'
+    )
+    choice(
+      choices: ['Do Not Deploy', 'dev', 'test', 'impl', 'prod'],
+      description: 'The environment to deploy to. Required.',
+      name: 'ENV'
+    )
+  }
+
+  stages {
+    stage('Ensure BB20_APP_VERSION, ENV, BB20_PLATINUM_AMI_ID and SUBNET_ID') {
+      steps {
+        sh """
+        if [ -z "${params.BB20_APP_VERSION}" ] || [ -z "${params.ENV}" ] || [ -z "${params.BB20_PLATINUM_AMI_ID}" ] || [ -z "${params.SUBNET_ID}" ]
+        then
+          exit 1
+        fi
+        """
+      }
+    }
+
+    stage('Stop master deployments to IMPL and PROD') {
+      steps {
+        sh """
+        if [ "${params.BB20_APP_VERSION}" == "master" ]
+        then
+          if [ "${params.ENV}" == "impl" ] || [ "${params.ENV}" == "prod" ]
+          then
+            exit 1
+          fi
+        fi
+        """
+      }
+    }
+
+    stage('Set release_version var') {
+      steps {
+        script {
+          if (params.BB20_APP_VERSION == 'develop' || params.BB20_APP_VERSION == 'master') {
+            release_version = "latest-${params.BB20_APP_VERSION}"
+          } else {
+            // Git tag, commit hash or branch other than master, develop
+            release_version = "${params.BB20_APP_VERSION}"
+          }
+        }
+      }
+    }
+
+/*
+# THIS IS WHERE WE WILL ADD SLACK NOTIFICATIONS ONCE SLACK IS ONLINE
+*/
+
+    stage('Set private key file') {
+      steps {
+        script {
+          private_key = "${params.ENV}-key"
+        }
+      }
+    }
+
+    stage('Checkout') {
+      steps {
+        checkout([
+          $class: 'GitSCM',
+          branches: [[
+            name: "${params.BB20_DEPLOY_BRANCH}"
+          ]],
+          doGenerateSubmoduleConfigurations: false,
```

Comment:
possibly need this to be `true` now that the openapi definition is pulled in through a submodule

Reply:
Nice catch. Thank you.